common/motion_utils/include/motion_utils/marker/marker_helper.hpp

```diff
@@ -41,6 +46,46 @@ visualization_msgs::msg::MarkerArray createDeletedSlowDownVirtualWallMarker(
 
 visualization_msgs::msg::MarkerArray createDeletedDeadLineVirtualWallMarker(
   const rclcpp::Time & now, const int32_t id);
+
+visualization_msgs::msg::MarkerArray createVirtualWallMarkerFromPreviousPoses(
+  const std::vector<Pose> & stop_poses, const std::vector<Pose> & previous_poses,
+  const rclcpp::Time & now, int32_t id);
+
+using create_wall_function = std::function<visualization_msgs::msg::MarkerArray(
+  const geometry_msgs::msg::Pose & pose, const std::string & module_name, const rclcpp::Time & now,
+  const int32_t id, const double longitudinal_offset)>;
+
+using delete_wall_function =
+  std::function<visualization_msgs::msg::MarkerArray(const rclcpp::Time & now, const int32_t id)>;
+
```

Comment:
Can you define `create_wall_function` and `delete_wall_function` in `VirtualWallMarkerCreator`:
```
class VirtualWallMarkerCreator {
   using create_wall_function = ...
   using delete_wall_function -= ...
}
```
to limit the scope of symbol ?

Reply:
Thanks!